packages/amplify-util-uibuilder/src/commands/utils/detachedFormCleanup.ts

```diff
@@ -0,0 +1,58 @@
+import { StudioForm, StudioSchema } from '@aws-amplify/codegen-ui';
+import { Form } from 'aws-sdk/clients/amplifyuibuilder';
+import { AmplifyStudioClient } from '../../clients';
+import { printer } from 'amplify-prompts';
+import asyncPool from 'tiny-async-pool';
+
+/**
+ * Does the form's reference DataStore type exist in the list of models
+ */
+export const isFormDetachedFromModel = (schema: StudioForm | Form, models: Set<string>): boolean => {
```

Comment:
I would also suggest that the models have stricter types instead of allowing a set of Strings and checking in the functions, create a ModelType that models the information better.

Reply:
models are names of models, they can only be strings. If I created a ModelType it would look like 
```
{name: string}
```
is that what you mean? If so, I'm not sure what the benefit is. What's wrong with allowing a set of strings?

Reply:
Perhaps a slight change in naming can clear up any confusion. Thoughts on changing `models` => `modelNames`?